simple_use_case/aiida/simple_use_case.py

```diff
@@ -0,0 +1,96 @@
+from pathlib import Path
+
+from aiida.orm import List, SinglefileData
+from aiida.engine.processes.functions import shellfunction, workfunction
+
+
+@shellfunction(command="gmsh", output_filenames=["mesh"])
+def gmsh(**kwargs):
+    """Run the ``gmsh`` command."""
+
+
+@shellfunction(command="meshio", output_filenames=["mesh.xdmf", "mesh.h5"])
+def meshio(**kwargs):
+    """Run the ``meshio`` command to convert to XDMF format."""
+
+
+@shellfunction(command="python", output_filenames=["*.pvd", "*.vtu"])
+def poisson(**kwargs):
+    """Run the ``poisson`` step."""
+
+
+@shellfunction(command="pvbatch", output_filenames=["plotoverline.csv"])
+def pvbatch(**kwargs):
+    """Run the ``pvbatch`` command."""
+
+
+@workfunction
+def workflow(geometry, poisson_script, post_processing_script):
+    """Simple workflow to solve Poisson equation for a geometry."""
+
+    # Generate the mesh from the geometry.
+    # Instead of writing to "mesh.msh" which would allow gmsh to deduce the output format
+    # we explicitly specify the format using the "-format" option.
+    # For certain reasons, the output link label in AiiDA would be converted to "mesh_msh"
+    # if we would have used "mesh.msh".
+    # For more information see section "Design choices" > "Output files" in the AiiDA
+    # Enhancement Proposal https://github.com/aiidateam/AEP/tree/aep/shell-functions/xxx_shell_functions
+    arguments = List(
+        ["-2", "-order", "1", "-format", "msh2", "{geometry}", "-o", "mesh"]
+    )
+    results_gmsh = gmsh(arguments=arguments, geometry=geometry)
+
+    # Convert the mesh to XDMF format
+    # The mesh file generated by gmsh is written to a temporary working directory
+    # with a temporary filename. Therefore, meshio can't deduce the input format
```

Comment:
I would mention that due to the above reasons the output does not have an ending that allows meshio to deduce the type. IMO it is not a temp. file (at least I found that first a little confusing).

Reply:
By now another commit was made, such that the actual filename is used. See this [line of code](https://github.com/sphuber/aiida-core/blob/feature/5287/shell-function/aiida/engine/processes/functions/shell.py#L177)
I will try to make this more intuitive by using `mesh.msh` instead of just `mesh` as the output filename.

Reply:
I was just refering to the documention, IMO you don't have to change the code (though any improvement is great)